test/modules/encoders/test_clip_text_encoder.py

```diff
@@ -0,0 +1,93 @@
+
+import pytest
+import torch
+from test.test_utils import assert_expected
+from torchmultimodal.modules.encoders.clip_text_encoder import CLIPTextEncoder
+from torchmultimodal.utils.common import get_current_device
+
+torch.manual_seed(1234)
+context_length = 77
+device = get_current_device()
+
+
+class TestCLIPTextEncoder:
+    def test_clip_init(self):
+        embedding_dim = 256
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), -143.8726, rtol=0, atol=1e-3)
+
+    def test_no_init(self):
+        embedding_dim = 256
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=False)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), 64.1976, rtol=0, atol=1e-3)
+
+    def test_small_embedding(self):
+        embedding_dim = 1
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), -1.4961, rtol=0, atol=1e-3)
+
+    def test_small_batch(self):
+        embedding_dim = 256
+        text = torch.randint(1, 79, (context_length,), dtype=torch.long).unsqueeze(0)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (1, embedding_dim)
+        assert_expected(out.sum().item(), 3.9050, rtol=0, atol=1e-3)
+
+    def test_under_context_length(self):
+        embedding_dim = 256
+        batch_size = 16
+        text = torch.randint(1, 79, (batch_size, context_length - 5), dtype=torch.long)
+
+        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
+        text_encoder = text_encoder.to(device)
+        assert isinstance(text_encoder, torch.nn.Module)
+
+        out = text_encoder(text)
+        assert out.shape == (batch_size, embedding_dim)
+        assert_expected(out.sum().item(), -142.1651, rtol=0, atol=1e-3)
```

Comment:
Similar comment to `test_clip_init`, this isn't really testing the fact that the input text is shorter than the context length. You could check that padding is occurring if you want to explicitly test this.